src/ComposerRequireChecker/DependencyGuesser/DependencyGuesser.php

```diff
@@ -26,4 +26,9 @@ public function __invoke(string $symbolName): Generator
             yield from $guesser($symbolName);
         }
     }
+
+    public function addGuesser(Guesser $guesser): void
```

Comment:
Let's avoid introducing mutability here: instead, let's use a new constructor argument?

Reply:
Optional or not: the BC break from using `roave/better-reflection` by default will probably be notable anyway, so we can potentially break BC here, and release in `5.0.0`

Reply:
Could you explain why adding the dependency would lead to a BC break?

Yeah, would go for the argument too.

Reply:
Potentially because scanning behavior changes => new symbols detected (or missed) => CI failure for consumers that upgrade.

Reply:
Maybe I misunderstand: But we are not detecting here more / new symbols, we just give here a hint which dependency should be installed to get rid of the warning, no?

Reply:
I removed the mutability but still i dont get why we already break BC with using better-reflection. Maybe you could explain it more?